chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/DeveloperManager.java

```diff
@@ -197,30 +196,38 @@ public DeveloperTree getHierarchyById(Long id) throws EntityRetrievalException {
 
         developerTree.getProducts().stream().forEach(product -> {
             product.getVersions().stream().forEach(version -> {
-                List<CertifiedProductSummaryDTO> versionListings = listings.stream()
+                List<SimpleListing> listingsForVersion = listings.stream()
                         .filter(listing -> listing.getVersion().getId().equals(version.getVersionId()))
+                        .map(listing -> convert(listing, acbs))
                         .collect(Collectors.toList());
-                versionListings.stream().forEach(listing -> {
-                    SimpleListing listingLeaf = new SimpleListing();
-                    Optional<CertificationBodyDTO> listingAcb = acbs.stream()
-                            .filter(acb -> acb.getId().equals(listing.getAcb().getId())).findFirst();
-                    if (listingAcb != null && listingAcb.isPresent()) {
-                        listingLeaf.setAcb(new CertificationBody(listingAcb.get()));
-                    }
-                    listingLeaf.setCertificationDate(listing.getCertificationDate().getTime());
-                    listingLeaf.setCertificationStatus(listing.getCertificationStatus());
-                    listingLeaf.setChplProductNumber(listing.getChplProductNumber());
-                    listingLeaf.setCuresUpdate(listing.getCuresUpdate());
-                    listingLeaf.setEdition(listing.getYear());
-                    listingLeaf.setId(listing.getId());
-                    listingLeaf.setLastModifiedDate(listing.getLastModifiedDate().getTime());
-                    version.getListings().add(listingLeaf);
+                    version.getListings().addAll(listingsForVersion);
                 });
             });
-        });
         return developerTree;
     }
 
+    private SimpleListing convert(CertifiedProductDetailsDTO listingDto, List<CertificationBodyDTO> acbs) {
```

Comment:
This is super minor and relates to Clean Code.  If you were to rename this function `convertToSimpleListing` it would make your `.map()` above a little more readable.  It relates to the concept of being clear with your function name as to what the purpose of the function is.  I'm only pointing this out because we talked about Clean Code a couple weeks ago.